frontends/mit-open/src/pages/HomePage/NewsEventsSection.tsx

```diff
@@ -0,0 +1,320 @@
+  background-size: cover;
+  background-repeat: no-repeat;
+  -webkit-background-position: center;
+  background-position: center;
+  width: 100%;
+  object-fit: cover;
+  height: 171px;
+  border-radius:;
+`
+
+const StoryTitle = styled.p`
+  ${{ ...theme.typography.subtitle1 }}
+  text-overflow: ellipsis;
+  height: ${theme.typography.pxToRem(40)};
+  overflow: hidden;
+  margin: 16px;
+
+  @supports (-webkit-line-clamp: 2) {
+    white-space: initial;
+    display: -webkit-box;
+    -webkit-line-clamp: 2;
+    -webkit-box-orient: vertical;
+  }
+`
+
+const StoryPublished = styled.span`
+  ${{
+    ...theme.typography.body3,
+    color: theme.custom.colors.silverGrayDark,
+  }}
+
+  margin: 0 16px 16px;
+`
+
+const Events = styled.div`
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  gap: 24px;
+  align-self: stretch;
+`
+
+const MobileEvents = styled(Events)`
+  padding: 0 16px;
+`
+
+const EventCard = styled(Card)`
+  display: flex;
+  align-items: center;
+  gap: 16px;
+  flex: 1 0 0;
+  align-self: stretch;
+  padding: 16px;
+  justify-content: space-between;
+`
+
+const EventDate = styled.div`
+  display: flex;
+  height: 64px;
+  flex-basis: 64px;
+  flex-shrink: 0;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  border-radius: 4px;
+  background: ${theme.custom.colors.lightGray1};
+`
+
+const EventDay = styled.p`
+  color: ${theme.custom.colors.red};
+  font-family: ${theme.typography.fontFamily};
+  font-size: ${theme.typography.pxToRem(28)};
+  font-weight: ${theme.typography.fontWeightBold};
+  line-height: ${theme.typography.pxToRem(36)};
+  margin: 0 0 -4px;
+`
+const EventMonth = styled.p`
+  margin: 0;
+  color: ${theme.custom.colors.silverGrayDark};
+  text-transform: uppercase;
+  ${{ ...theme.typography.subtitle3 }}
+`
+
+const EventTitle = styled.p`
+  height: ${theme.typography.pxToRem(59)};
+  align-self: stretch;
+  color: ${theme.custom.colors.darkGray2};
+  ${{ ...theme.typography.subtitle1 }}
+  margin: 0;
+  overflow: hidden;
+  margin-right: auto;
+
+  @supports (-webkit-line-clamp: 3) {
+    white-space: initial;
+    display: -webkit-box;
+    -webkit-line-clamp: 3;
+    -webkit-box-orient: vertical;
+  }
+`
+
+const Chevron = styled(RiArrowRightSLine)`
+  width: 24px;
+  height: 24px;
+  flex-shrink: 0;
+  fill: ${theme.custom.colors.silverGray};
+  justify-content: flex-end;
+`
+
+const Story: React.FC<{ item: NewsFeedItem; mobile: boolean }> = ({
+  item,
+  mobile,
+}) => {
+  return (
+    <StoryCard mobile={mobile} href={item.url}>
+      <StoryImage src={item.image?.url} alt={item.image?.alt} />
+      <StoryTitle>{item.title}</StoryTitle>
+      <StoryPublished>
+        Published: {formatDate(item.news_details?.publish_date)}
+      </StoryPublished>
+    </StoryCard>
+  )
+}
+
+const NewsEventsSection: React.FC = () => {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const { data: news }: any = useNewsEventsList({
```

Comment:
Seeing an issue with these with `Type 'FeedItem' is not assignable to type 'NewsFeedItem'.`

The return type on the `useNewsEventsList()` query does not pick up on the discriminator https://github.com/mitodl/mit-open/blob/6bbdf4abaa160cd5b16dd530d4792a9aedc244c7/frontends/api/src/generated/v0/api.ts#L618-L620

_Possibly_ due to discrepancy between `feed_type` on the request params and `resource_type` on the `FeedItem`?

Reply:
That worked great, thanks @gumaerc. 

The only issue now is that the change to the `PaginatedFeedItemList` interface is in the generated API code. Do you know how we can make that change in the python source?
https://github.com/mitodl/mit-open/blob/6bbdf4abaa160cd5b16dd530d4792a9aedc244c7/frontends/api/src/generated/v0/api.ts#L1225